rancher-sandbox/rancher-desktop
# Implement port server over named pipe

To test:

Run the host agent using [this](https://github.com/rancher-sandbox/rancher-desktop-agent/pull/8#issue-1365288460) instructions.

Build and run the Privileges service. 

```
GOOS=windows go build .
```

Run
```
.\privileged-service.exe install
.\privileged-service.exe start
```

Run a docker command with published port:

```
docker run -d --name test-nginx2 -p 38083:80 nginx
```

Verify netstat and netsh

```
 netstat -ant |findstr '38083'
  TCP    0.0.0.0:38083          0.0.0.0:0              LISTENING       InHost
  TCP    127.0.0.1:38083        0.0.0.0:0              LISTENING       InHost
  TCP    [::]:38083             [::]:0                 LISTENING       InHost
  TCP    [::1]:38083            [::]:0                 LISTENING       InHost
```

Netsh
```
netsh interface portproxy show all

Listen on ipv4:             Connect to ipv4:

Address         Port        Address         Port
--------------- ----------  --------------- ----------
0.0.0.0         38083       172.18.106.220  80

Listen on ipv6:             Connect to ipv6:

Address         Port        Address         Port
--------------- ----------  --------------- ----------
::              38083       fe80::215:5dff:fef3:fa06 80
```

Remove the port mapping
```
docker stop containerID
docker rm containerID
```

Checkt the netstat and netsh again to verify entries are removed.
